### PR TITLE
Added a little Async Magic

### PR DESCRIPTION
**Pull request by Aniel**
GetSubreddit Async

### CODE REVIEW

File: RedditSharp/Things/Thing.cs [hunk 1]
```diff
@@ -74,5 +75,38 @@ public TimeSpan TimeSinceFetch
...
+            Thing result = await ParseAsync(reddit, json, webAgent);
+            if (result == null)
+            {
+                if (typeof(T) == typeof(WikiPageRevision))
+                {
+                }
+            }
+            return result;
+        }
     }
```
Comment (SirCmpwn) [inline RedditSharp/Things/Thing.cs]: Is this finished?

Comment (Aniel) [inline RedditSharp/Things/Thing.cs]: I think. GetSubredditAsync works for me great so far.

Comment (SirCmpwn) [inline RedditSharp/Things/Thing.cs]: What is it meant to do? Lines 103 to 108 look useless.

Comment (Aniel) [inline RedditSharp/Things/Thing.cs]: Oh thank you vor pointing out there should be this piece of Code inside the if:
return await new WikiPageRevision().InitAsync(reddit, json, webAgent);
Do i Need to add a new pull request or can i add to this?

Comment (SirCmpwn) [inline RedditSharp/Things/Thing.cs]: You can simply add it to this.

**Comment by SirCmpwn** [comment]
Looks good. Thanks!

**Pull request merged**